bin/faucet/src/errors.rs

```diff
@@ -21,27 +20,34 @@ pub enum FaucetError {
 
     #[error("Failed to create Miden account: {0}")]
     AccountCreationError(String),
-}
-
-impl error::ResponseError for FaucetError {
-    fn error_response(&self) -> HttpResponse<actix_web::body::BoxBody> {
-        let message = match self {
-            FaucetError::StartError(msg) => msg.to_string(),
-            FaucetError::BadRequest(msg) => msg.to_string(),
-            FaucetError::ConfigurationError(msg) => msg.to_string(),
-            FaucetError::InternalServerError(msg) => msg.to_string(),
-            FaucetError::AccountCreationError(msg) => msg.to_string(),
-        };
 
-        HttpResponse::build(self.status_code())
-            .insert_header(ContentType::html())
-            .body(message.to_owned())
-    }
+    #[error("Page not found: {0}")]
+    NotFound(String),
+}
 
-    fn status_code(&self) -> actix_web::http::StatusCode {
+impl FaucetError {
+    fn status_code(&self) -> StatusCode {
         match *self {
             FaucetError::BadRequest(_) => StatusCode::BAD_REQUEST,
+            FaucetError::NotFound(_) => StatusCode::NOT_FOUND,
             _ => StatusCode::INTERNAL_SERVER_ERROR,
         }
     }
 }
+
+impl IntoResponse for FaucetError {
+    fn into_response(self) -> Response {
+        let status_code = self.status_code();
+        let message = match self {
+            FaucetError::StartError(msg) => msg,
+            FaucetError::BadRequest(msg) => msg,
+            FaucetError::ConfigurationError(msg) => msg,
+            FaucetError::InternalServerError(msg) => msg,
+            FaucetError::AccountCreationError(msg) => msg,
+            FaucetError::NotFound(msg) => msg,
+        };
```

Comment:
While we're here - are there any errors we shouldn't be exposing to the caller?

`StartError` and `ConfigurationError` sound like they should be impossible to return to the caller. `InternalError` is also exposing the error string which is probably a bad idea.

Reply:
Yes, I agree with you. I'd rather separate this enum to two different. Will do it, thanks!